### PR TITLE
Remove parsers

### PR DESCRIPTION
**Pull request by ItzikAlkotzerAidoc**
Remove the parsers.
instead we use the creators, to parse the json to model.
keep list of models, in the jsonToFormWidget
creating widget per model (by the type) when required.
-- this cause deprecation of the parser factory.

### CODE REVIEW

File: lib/json_to_form_with_theme.dart [hunk 1]
```diff
@@ -9,51 +9,106 @@ import 'package:json_to_form_with_theme/exceptions/parsing_exception.dart';
...
+  JsonFormWithTheme._({
     Key? key,
     required this.jsonWidgets,
     this.onValueChanged,
-    this.dynamicFactory,
     this.theme = const DefaultTheme(),
     this.streamUpdates,
     this.dateBuilder,
-  }) : super(key: key);
+  }) :_parsersCreateors = HashMap(), super(key: key);
```
Comment (itaishalom) [inline lib/json_to_form_with_theme.dart:97]: typo

File: lib/json_to_form_with_theme.dart [hunk 2]
```diff
@@ -9,51 +9,106 @@ import 'package:json_to_form_with_theme/exceptions/parsing_exception.dart';
...
   @override
   _JsonFormWithThemeState createState() => _JsonFormWithThemeState();
 }
 
 class _JsonFormWithThemeState extends State<JsonFormWithTheme> {
-  HashMap<String, WidgetParser> parsers = HashMap();
-  List<Widget> widgetsGlobal = [];
   late final StreamSubscription<Map<String, dynamic>>? _valueChange;
+  late Stream<DataClass> dataClassStream;
+  bool ignoreRebuild = false;
```
Comment (itaishalom) [inline lib/json_to_form_with_theme.dart:106]: check if can be deleted

File: lib/json_to_form_with_theme.dart [hunk 3]
```diff
@@ -197,9 +196,10 @@ class _JsonFormWithThemeState extends State<JsonFormWithTheme> {
                       SliverList(
                         delegate: SliverChildBuilderDelegate(
                           (BuildContext context, int index) {
-                            return widgetsGlobal[index];
+                            ItemModel item = widget.items[index];
+                            return (widget._parsersCreateors[item.type]?? EmptyCreator()).createWidget(item, widget.onValueChanged,  widget.dateBuilder);
```
Comment (itaishalom) [inline lib/json_to_form_with_theme.dart:200]: Change name

File: lib/widgets/drop_down_widget.dart [hunk 4]
```diff
@@ -11,37 +12,41 @@ import 'name_description_widget.dart';
...
+  // final String? description;
+  // final String id;
+  // final List<String> values;
+  // String? chosenValue;
+  // final bool isBeforeHeader;
+  final Widget Function(int date, String id)? dateBuilder;
+  // int? time;
+  final DropDownModel model;
+  final OnValueChanged? onValueChanged;
+
```
Comment (itaishalom) [inline lib/widgets/drop_down_widget.dart:45]: delete comments

File: lib/json_to_form_with_theme.dart (hunk 3 above)
Comment (ItzikAlkotzerAidoc) [reply lib/json_to_form_with_theme.dart:200]: done

File: lib/widgets/drop_down_widget.dart (hunk 4 above)
Comment (ItzikAlkotzerAidoc) [reply lib/widgets/drop_down_widget.dart:45]: done

File: lib/json_to_form_with_theme.dart (hunk 1 above)
Comment (ItzikAlkotzerAidoc) [reply lib/json_to_form_with_theme.dart:97]: fixed

**Pull request merged**